### ISSUE TITLE
Python layer using lsp backend: (void-variable get-root)

### ISSUE THREAD

**Issue by fleimgruber** [syl20bnr/spacemacs#10391]
#### Description :octocat:
Python layer with backend `lsp` errors on unassigned symbol `get-root`.

#### Reproduction guide :beetle:
- Start Emacs
- Visit a Python `.py` file

*Observed behaviour:* :eyes: :broken_heart:
Error and traceback below.

*Expected behaviour:* :heart: :smile:
Python layer operational with `lsp-python` enabled.

#### System Info :computer:
- OS: darwin
- Emacs: 26.0.91
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 671701f2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(pbz lsp semantic-web javascript helm bibtex auto-completion dash sphinx restructuredtext emacs-lisp
     (git :variables git-enable-magit-svn-plugin t)
     rust clojure ess
     (python :variables python-test-runner 'pytest python-backend 'lsp)
     html
     (markdown :variables markdown-command "pandoc")
     (windows-scripts :toggle
                      (eq system-type 'windows-nt))
     (shell-scripts :toggle
                    (or
                     (eq system-type 'darwin)
                     (eq system-type 'gnu)))
     (org :variables org-enable-reveal-js-support t)
     syntax-checking
     (c-c++ :variables c-c++-enable-clang-support t)
     (osx))
```
- System configuration features: JPEG RSVG IMAGEMAGICK NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS LCMS2

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-variable get-root)
  lsp-python-enable()
  (progn (require 'lsp-python) (lsp-python-enable))
  (if (configuration-layer/layer-used-p 'lsp) (progn (require 'lsp-python) (lsp-python-enable)) (message "`lsp' layer is not installed, please add `lsp' layer to your dofile."))
  spacemacs//python-setup-lsp()
  (cond ((eq python-backend 'anaconda) (spacemacs//python-setup-anaconda)) ((eq python-backend 'lsp) (spacemacs//python-setup-lsp)) (t nil))
  spacemacs//python-setup-backend()
  run-hooks(change-major-mode-after-body-hook prog-mode-hook python-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook prog-mode-hook python-mode-hook))
  run-mode-hooks(python-mode-hook)
  python-mode()
  set-auto-mode-0(python-mode nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
  find-file-noselect-1(#<buffer indexing.py> "~/dev/pandas/pandas/core/indexing.py" nil nil "~/dev/pandas/pandas/core/indexing.py" (3140299 16777220))
  find-file-noselect
```

**Comment by arejensen** [comment]
[This lsp-mode fix](https://github.com/emacs-lsp/lsp-mode/pull/296) seems to work for me. I am still experiencing other issues afterwards, but that'll go in a different bug-report.

**Comment by fleimgruber** [comment]
@arejensen thanks for the pointer.  https://github.com/emacs-lsp/lsp-mode/pull/296 worked for me.

**Comment by non-Jedi** [comment]
Upstream issue was emacs-lsp/lsp-mode#295 and appears to have been closed without emacs-lsp/lsp-mode#296. I think this spacemacs bug can be closed now; haven't run into the problem since emacs-lsp/lsp-mode#295 was closed.

**Comment by vv111y** [comment]
I filed https://github.com/syl20bnr/spacemacs/issues/10449. Same prob even after fix https://github.com/emacs-lsp/lsp-mode/commit/12630a1e39c0aa66e12d32051378399a750e9fdb
